app/models/patient_session.rb

```diff
@@ -92,7 +93,8 @@ def able_to_vaccinate?
   end
 
   def safe_to_destroy?
-    vaccination_records.empty? && gillick_assessment.nil?
+    vaccination_records.empty? && gillick_assessment.nil? &&
+      session_attendances.none?(&:attending?)
```

Comment:
I think this is the right logic but I think we'll need to add something to `destroy_if_safe!` or the relationship to make sure that if we do try and destroy this patient session and there are only session attendances where `attending = false` they get deleted as well, I think currently we'll get a foreign key constraint error.

Reply:
We can probably use `dependent:` again 🤔 Good catch